src/conf_mode/vpp_nat_static.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2025 VyOS Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+from vyos.config import Config
+from vyos import ConfigError
+from vyos.vpp.nat.nat44 import Nat44Static
+
+
+protocol_map = {
+    'all': 0,
+    'icmp': 1,
+    'tcp': 6,
+    'udp': 17,
+}
+
+
+def get_config(config=None) -> dict:
+    if config:
+        conf = config
+    else:
+        conf = Config()
+
+    base = ['vpp', 'nat44', 'static']
+
+    # Get config_dict with default values
+    config = conf.get_config_dict(
+        base,
+        key_mangling=('-', '_'),
+        get_first_key=True,
+        no_tag_node_value_mangle=True,
+        with_defaults=True,
+        with_recursive_defaults=True,
+    )
+
+    # Get effective config as we need full dictionary per interface delete
+    effective_config = conf.get_config_dict(
+        base,
+        key_mangling=('-', '_'),
+        effective=True,
+        get_first_key=True,
+        no_tag_node_value_mangle=True,
+    )
+
+    if not config:
+        config['remove'] = True
+
+    if effective_config:
+        config.update({'effective': effective_config})
+
+    return config
+
+
+def verify(config):
+    if 'remove' in config:
+        return None
+
+    for rule, rule_config in config['rule'].items():
+        required_keys = {'inbound_interface', 'outbound_interface'}
+        if not all(key in rule_config for key in required_keys):
+            missing_keys = required_keys - set(rule_config.keys())
+            raise ConfigError(
+                f"Required options are missing: {', '.join(missing_keys).replace('_', '-')} in rule {rule}"
+            )
```

Comment:
The logic is a bit overcomplicated and non-optimal.
I would prefer something like:

```
required_keys = {'inbound_interface', 'outbound_interface'}
for rule, rule_config in config['rule'].items():
    missing_keys = required_keys - rule_config.keys()
    if missing_keys:
        raise ConfigError(
            f"Required options are missing: {', '.join(missing_keys).replace('_', '-')} in rule {rule}"
        )
```